velox/functions/sparksql/String.h

```diff
@@ -1130,4 +1130,52 @@ struct ReplaceFunction {
   }
 };
 
+template <typename T>
+struct FindInSetFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<int32_t>& result,
+      const arg_type<Varchar>& str,
+      const arg_type<Varchar>& strArray) {
+    int32_t index = 1;
+    int32_t lastComma = -1;
+
+    if (std::string_view(str).find(',') != std::string::npos) {
+      result = 0;
+      return;
+    }
+
+    auto arrayData = strArray.data();
+    size_t arraySize = strArray.size();
+    size_t matchSize = str.size();
+
+    for (int i = 0; i < arraySize; i++) {
```

Comment:
why not use std::string_view::find?

https://en.cppreference.com/w/cpp/string/basic_string_view/find

Reply:
If this is the case, the find method may not find the correct query
`find_in_set("ab","abcdefg,ab,hh")`

Reply:
I mean why not use std::string_view::find to find next comma.

Reply:
I know what you mean, but I'm not sure which of the two approaches is better, The reason I chose this approach is to keep it consistent with the spark function. Thanks.

Reply:
I suggest to write a small benchmark to see if there is any perf difference.

See velox/benchmarks/basic/LikeBenchmark.cpp for an example.

Reply:
see https://github.com/apache/spark/blob/667c0a9dbbe045c73842a345c1b3897b155564d4/common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java#L471

Reply:
I think we should keep current logic, I did three sets of tests, and I found that the while loop way performed better than the `std::string_view::find` method, Here are some of my test results.` test goup1` tests to find the substring at the front of the strArray,` test group 2` at the middle, and `test goup 3` at the end. As the substring appears further back, the performance of the method using string_view::find decreases more. Thanks.

`test group 1`:
while loop cost time : 	5.7535s
while loop cost time : 	5.97309s
while loop cost time : 	5.71566s
string_view::find  cost time : 	5.96654s
string_view::find  cost time : 	5.97334s
string_view::find  cost time : 	5.97478s


`test group test2`:
while loop cost time : 	9.72591s
while loop cost time : 	9.78818s
while loop cost time : 	9.76955s
string_view::find  cost time : 	14.807s
string_view::find  cost time : 	14.8721s
string_view::find  cost time : 	14.785s

`test group test3`:
while loop cost time : 	19.2678s
while loop cost time : 	18.9274s
while loop cost time : 	18.7739s
string_view::find  cost time : 	27.0144s
string_view::find  cost time : 	27.2736s
string_view::find  cost time : 	27.4933s

Reply:
In my tests, `string_view::find` performs better when determining whether a substring contains a comma, and the for loop performs better when finding comma indexs. I think this is the best combination. Thanks!

Reply:
Got it. Thank you for checking.